test/jdk/javax/swing/JTable/KeyBoardNavigation.java

```diff
@@ -146,25 +139,66 @@ public void setValue(Object value) {
         numbersColumn.setCellRenderer(numberColumnRenderer);
         numbersColumn.setPreferredWidth(110);
 
-        // Finish setting up the table.
-        JScrollPane scrollpane = new JScrollPane(tableView);
-        scrollpane.setBorder(new BevelBorder(BevelBorder.LOWERED));
-        scrollpane.setPreferredSize(new Dimension(430, 200));
+        JScrollPane scrollPane = new JScrollPane(tableView);
+        scrollPane.setBorder(new BevelBorder(BevelBorder.LOWERED));
+        scrollPane.setPreferredSize(new Dimension(430, 200));
 
-        contentPane.add(scrollpane);
+        frame.add(scrollPane);
+        frame.pack();
+        frame.setVisible(true);
     }
 
+    private static JTable getTableDetails(String[] names, Object[][] data) {
+        TableModel dataModel = new AbstractTableModel() {
+            // These methods always need to be implemented.
+            public int getColumnCount() {
+                return names.length;
+            }
+
+            public int getRowCount() {
+                return data.length;
+            }
+
+            public Object getValueAt(int row, int col) {
+                return data[row][col];
+            }
+
+            // The default implementations of these methods in
+            // AbstractTableModel would work, but we can refine them.
+            public String getColumnName(int column) {
+                return names[column];
+            }
+
+            public Class<?> getColumnClass(int c) {
+                return getValueAt(0, c).getClass();
+            }
+
+            public boolean isCellEditable(int row, int col) {
+                return true;
+            }
 
-    public void init() {
-        SwingUtilities.invokeLater(() -> {
+            public void setValueAt(Object aValue, int row, int column) {
+                System.out.println("Setting value to: " + aValue);
+                data[row][column] = aValue;
+            }
+        };
+
+        JTable tableView = new JTable(dataModel);
+        // Turn off auto-resizing so that we can set column sizes programmatically.
+        // In this mode, all columns will get their preferred widths, as set blow.
+        tableView.setAutoResizeMode(JTable.AUTO_RESIZE_OFF);
+        return tableView;
+    }
+
+    public static void main(String[] args) throws Exception {
+        UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
```

Comment:
The fix is for all L&F but the test is testing only Metal...It should be given chance to use default L&F in each platform..
If 4112270 is only about Metal, then a new test should be spawned to test this fix for all fixed L&F..

Reply:
4112270 is not only about Metal L&F. In Aqua L&F the functionality is different, it enables the cell to edit mode. Yet, in macos excel functionality is same as that of windows. Should we update Aqua too?

Reply:
Which macos excel application you tested? Microsoft excel or NUmbers...I guess Numbers should be ideal native application to tell if native OS does need it and depending on that, we need to change it accordingly for Aqua L&F..

Reply:
In both Microsoft excel and Numbers, Ctrl shift UP/DOWN extends till first/end row.